src/components/results/loadflow/load-flow-result-utils.ts

```diff
@@ -186,68 +190,74 @@ export const makeData = (
 };
 
 // We can use this custom hook for fetching enums for AutoComplete filter
-export const useFetchFiltersEnums = (
-    studyUuid: UUID,
-    nodeUuid: UUID,
-    hasResult: boolean = false,
-    setFilter: (value: boolean) => void
-): { error: boolean; loading: boolean; result: FilterEnumsType } => {
+export const useFetchFiltersEnums = (): {
+    error: boolean;
+    loading: boolean;
+    result: FilterEnumsType;
+} => {
     const [loading, setLoading] = useState(true);
     const [error, setError] = useState(false);
     const [result, setResult] = useState<FilterEnumsType>({
         status: null,
         limitType: null,
         side: null,
     });
+    const studyUuid = useSelector((state: ReduxState) => state.studyUuid);
+    const currentNode = useSelector(
+        (state: ReduxState) => state.currentTreeNode
+    );
+    const loadFlowStatus = useSelector(
+        (state: ReduxState) => state.computingStatus[ComputingType.LOAD_FLOW]
+    );
 
     useEffect(() => {
-        if (!hasResult) {
-            const promises = [
-                // We can add another fetch for other enums
-                fetchAvailableFilterEnumValues(
-                    studyUuid,
-                    nodeUuid,
-                    computingType.LOAD_FLOW,
-                    'computation-status'
-                ),
-                fetchAvailableFilterEnumValues(
-                    studyUuid,
-                    nodeUuid,
-                    computingType.LOAD_FLOW,
-                    'limit-types'
-                ),
-                fetchAvailableFilterEnumValues(
-                    studyUuid,
-                    nodeUuid,
-                    computingType.LOAD_FLOW,
-                    'branch-sides'
-                ),
-            ];
-
-            setLoading(true);
-            Promise.all(promises)
-                .then(
-                    ([
-                        computationsStatusResult,
-                        limitTypesResult,
-                        branchSidesResult,
-                    ]) => {
-                        setResult({
-                            status: computationsStatusResult,
-                            limitType: limitTypesResult,
-                            side: branchSidesResult,
-                        });
-                        setFilter(true);
-                        setLoading(false);
-                    }
-                )
-                .catch((err) => {
-                    setFilter(false);
-                    setError(err);
-                    setLoading(false);
-                });
+        if (loadFlowStatus !== RunningStatus.SUCCEED) {
+            return;
         }
-    }, [hasResult, setFilter, studyUuid, nodeUuid]);
+        const promises = [
```

Comment:
to reduce redundancy and make the  fetchAvailableFilterEnumValues function more maintainable, we can simplify the code by :

```
`const filterTypes = ['computation-status', 'limit-types', 'branch-sides'];

const promises = filterTypes.map(type =>
    fetchAvailableFilterEnumValues(studyUuid, currentNode.id, computingType.SECURITY_ANALYSIS, type)
);`
```

Reply:
fixed